pyperf/_utils.py

```diff
@@ -317,9 +320,33 @@ def __exit__(self, *args):
 class ReadPipe(_Pipe):
     def open_text(self):
         file = open(self._fd, "r", encoding="utf8")
+        if self._timeout:
+            os.set_blocking(file.fileno(), False)
         self._file = file
         return file
 
+    def read_text(self):
+        with self.open_text() as rfile:
+            if self._timeout is not None:
+                return self._read_text_timeout(rfile, self._timeout)
+            else:
+                return rfile.read()
```

Comment:
I don't think that io.TextIOWrapper, created by open_text(), handles well non-blocking file, and so I would prefer to work on "raw" file descriptors:

```suggestion
        if self._timeout is not None:
            return self._read_text_timeout(self._timeout)
        else:
            with self.open_text() as rfile:
                return rfile.read()
```